core/blockchain.go

```diff
@@ -987,6 +987,7 @@ func (bc *BlockChain) procFutureBlocks() {
 			if i == len(blocks)-1 && err == nil {
 				engine, ok := bc.Engine().(*XDPoS.XDPoS)
 				if ok {
+					i := i
```

Comment:
could you use other variable, i := i will work but make unnecessary confusion when someone looks at this code.